raiden/network/proxies/service_registry.py

```diff
@@ -45,19 +53,36 @@ def __init__(
         self.client = jsonrpc_client
         self.node_address = self.client.address
 
-    def service_count(self, block_identifier: BlockSpecification) -> int:
-        """Get the number of registered services"""
-        result = self.proxy.contract.functions.serviceCount().call(
+    def ever_made_deposits(
+        self, block_identifier: BlockSpecification, index: int
+    ) -> Optional[Address]:
+        """Get one of the addresses that have ever made a deposit."""
+        try:
+            ret = Address(
+                to_canonical_address(
+                    self.proxy.contract.functions.ever_made_deposits(index).call(
+                        block_identifier=block_identifier
+                    )
+                )
+            )
+            assert isinstance(ret, T_Address)
+            return ret
+        except BadFunctionCallOutput:
+            return None
+
+    def ever_made_deposits_len(self, block_identifier: BlockSpecification) -> int:
+        """Get the number of addresses that have ever made a deposit"""
+        result = self.proxy.contract.functions.everMadeDepositsLen().call(
             block_identifier=block_identifier
         )
         return result
 
-    def get_service_address(
-        self, block_identifier: BlockSpecification, index: int
-    ) -> Optional[AddressHex]:
-        """Gets the address of a service by index. If index is out of range return None"""
+    def has_valid_registration(
+        self, block_identifier: BlockSpecification, address: Address
+    ) -> Optional[bool]:
         try:
-            result = self.proxy.contract.functions.service_addresses(index).call(
+            assert isinstance(address, T_Address)
```

Comment:
same here. I guess we wanted to not have runtime type assertions on every function, that's why we switched to mypy.

Reply:
The same here (locally my `mypy` was not working).  Will remove.